MyFridge/MyFridge/Shelf.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace MyFridge
+{
+    internal class Shelf
+    {
+        public static int UniqueId { get; set; } = 1;
+        public int Id { get; }
+        public int FloorNumber { get; }
+        private int _spaceAvailable = 400;
+
+        public int SpaceAvailable
+        {
+            get => _spaceAvailable; set
+            {
+                if (value <= 0)
+                {
+                    throw new Exception("There is no space");
+                }
+
+                _spaceAvailable = value;
+            }
+        }
+        public List<Item> Items { get; set;}
+
+
+        public Shelf(int floorNumber, List<Item> items)
+        {
+            
+            Id += UniqueId++;
+
+            FloorNumber = floorNumber;
+                //if (items != null)
+                //{
+                //    foreach (var item in items)
+                //    {
+                //        if (item == null) items.Remove(item);
+                //    }
+                foreach (var item in items)
+                {
+                    if (_spaceAvailable - item.Space >= 0)
+                        _spaceAvailable -= item.Space;
+                    else
+                        items.Remove(item);
+                }
+            if (items.Count > 0)
+                Items = items;
+            else
+                throw new Exception("The items are not initialized");
```

Comment:
always try to help the user with information. for example: 
"Shelf could not be initialized. Shelf size: {..}, total items size: {..}"
This way the user all the information to retry and succeed the next time